### ISSUE TITLE
Allow url more than 1024 bytes for drm channels (widevine license url)

### ISSUE THREAD

**Issue by aicynide** [kodi-pvr/pvr.iptvsimple#776]

**Comment by phunkyfish** [comment]
No, I’m afraid this is not possible. The PVR API is limited to transferring 1024 bytes. In this case a video addon should be created to allow sensible URLs be used using `plugin://` URLS.

**Comment by aicynide** [comment]
Who made this "future-proof" api?
>Kodi can play anything

**Comment by phunkyfish** [comment]
Kodi can play anything. PVR not so much. The problem with changing the size for the URLs is that 32kb is far too much to pass between Kodi Core and the binary add-on (they must translate between C and C++). Change the value requires a rebuild of all of the add-ons which is a lot of work and can only be done for major releases of kodi.

Note that PVR was never designed to handle long URLs, it was designed to work with PVR backends which handle any of that complexity. Iptvsimple is different in that it doesn’t have a backend, it’s just has an M3U file.
